File: test/data/cpp/test06/StateProcessData.cpp
#include "StateProcessData.h"
StateProcessData::StateProcessData(StateController* stateController, IControllee* controllee) : BaseState(controllee)
{
    _stateController = stateController;
    _controllee = controllee;
}
StateProcessData::~StateProcessData()
{
}
const char* StateProcessData::GetStateName()
{
    return "State3.State4.ProcessData";
}
BaseState* StateProcessData::GetParent()
{
    return _stateController->InstanceOfState4;
}

File: test/data/cs/test06/StateProcessData.cs
public class StateProcessData : BaseState
{
    private StateController _stateController;
    private IControllee _controllee;
    public StateProcessData(StateController stateController, IControllee controllee) : base(controllee)
    {
        _stateController = stateController;
        _controllee = controllee;
    }
    public override string GetStateName()
    {
        return "State3.State4.ProcessData";
    }
    public override BaseState? GetParent()
    {
        return _stateController.InstanceOfState4;
    }
    public override int GetStateID()
    {
        return 0;
    }
}